src/main/java/bio/terra/app/logging/PerformanceLoggerON.java

```diff
@@ -0,0 +1,128 @@
+package bio.terra.app.logging;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Primary;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import java.io.Serializable;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+@Primary
+@Profile({"perftest"})
+@Component
+
+/**
+ * This class implements ENABLED performance logging. It is used when the "perftest" profile is active.
+ *
+ * It generates logs using the same slf4j logger used throughout the API code, but with a specific
+ * format intended to be propagated from StackDriver to BigQuery and then parsed with SQL for further analysis.
+ *
+ * Any time there is a new type of information that we want to include in performance logs, there are 2 options:
+ *   - Add the new field to the format string in this class and add optional arguments to the log methods that accept
+ *     a value for this new field.
+ *   - Specify the new information as a serializable object (e.g. String or property of a POJO) and pass it to the
+ *     existing log methods with the additionalInfo parameter.
+ *  New types of information that seem broadly useful should use the first option and those that will probably only
+ *  be used in a small number of places should use the second option.
+ */
+public class PerformanceLoggerON implements PerformanceLogger {
+
+    public static final boolean isPerformanceLoggingEnabled = true;
+
+    private ObjectMapper objectMapper;
+
+    private static Logger logger = LoggerFactory.getLogger(PerformanceLoggerON.class);
+    private static final String PerformanceLogFormat =
```

Comment:
How does a component downstream recognize this format as a performance log entry?
I would have expected some prefix used for filtering.

Reply:
Initially I used a prefix "PerformanceLogger, ". But these all are printed with the same format as our regular logger.info calls. That is, they are prefixed with the class name, which in this case is "PerformanceLoggerON" (example below). For this reason, I didn't add a duplicate prefix.

2020-06-23 11:06:18,875 INFO  [pool-6-thread-1] b.t.app.logging.PerformanceLoggerON: TimestampUTC: 2020-06-23T15:06:18.875Z, JobId: q62fqitASXK_iX4R_j7HZA, Class: bio.terra.service.dataset.flight.create.DatasetCreateFlight, Operation: startFlight, ElapsedTime: PT0S, IntegerCount: 0, AdditionalInfo: